README.md

```diff
@@ -1,81 +1,164 @@
+* Type `./eeprom-fixer.py savefile.sav`, where `savefile.sav` is the path to your `.SAV` file.
+
+The steps for Linux may work on macOS with [Python 3](https://www.python.org/downloads/mac-osx/) installed, but have not been tested.
+
+Your save file should now be fixed and ready to use with open_agb_firm. The script also works vise versa, if you want to use a save generated by open_agb_firm with an emulator.
+
+## FAQ
+**Q: Why isn't open_agb_firm a normal 3DS app?**\
+A: To access the 3DS's GBA hardware, open_agb_firm needs to run with full hardware access, which can only be provided by running as a FIRM.
+
+**Q: Is this safe to use?**\
+A: Of course! While open_agb_firm does run with full hardware access, a lot of work has been put in by several people to ensure that nothing unexpected happens. Some backend code from open_agb_firm is actually used in [fastboot3ds](https://github.com/derrekr/fastboot3DS)!
+
+**Q: What games work with open_agb_firm?**\
+A: In theory, all of them, except those that fall within the [hardware limitations](#hardware-limitations).
+
+**Q: How can I increase the brightness?**\
+A: Increase the value of the `backlight` setting in `config.ini`. See [Configuration](#configuration) for more information.
+
+**Q: Why do the colors look darker?**\
+A: The default gamma settings simulate the OG GBA's LCD. If you want the raw colors (basically what you see on emulators), set the `adjustGamma` setting to `false`. Do note that a lot of games were designed with a darker gamma in mind, so some games may look weird with the brighter colors.
+
+**Q: Why do some of my ROM hacks/homebrew games have saving issues?**\
+A: open_agb_firm resorts to save autodetection when it can't find an entry for the game it's running in `gba_db.bin` (which only contains data for official games), and it's a bit wonky for games that use EEPROM or misleading SDK save strings.
+
+**Q: Why doesn't my save file from an emulator work?**\
+A: There's a good chance that the save you're having issues with is an EEPROM save, which most emulators output differently. See [EEPROM Fixer](#eeprom-fixer).
+
+**Q: My game doesn't save properly!**\
+A: First, please ensure that the GBA ROM you are playing is not modified in any way, and matches its [No-Intro](https://datomatic.no-intro.org/) checksums. Second, make sure you aren't using an existing `.SAV` file, because some may have issues for various reasons. Third, make sure your [`gba_db.bin`](resources/gba_db.bin) is up-to-date. If everything seems to be in order but the game still doesn't save properly, please [open an issue](https://github.com/profi200/open_agb_firm/issues) so it can be fixed. In the meantime, the `useGbaDb` and `saveOverride` settings may be useful (see [Configuration](#configuration) for more information).
+
+## Compiling
+If you're using Windows 10, you can install and perform the following steps using [WSL 2](https://docs.microsoft.com/en-us/windows/wsl/install-win10).
```

Comment:
Forgive me if I'm wrong, but aren't the tools and the Makefile designed for Linux? I noticed that little things in the Makefile like the `7z` command would probably break on Windows, since when 7-Zip is installed, it's under `7za.exe`.

Reply:
The 7-Zip stuff is actually something i added myself to the makefile. I'm not sure how much it really relies on a Linux-like shell. When you use their installer you will see it installs msys2 with the devkitPro repo pre-installed so then you can add packages easily. WSL works fine too but it is not the recommended way.

Reply:
I know devkitPro works with MSYS2, but what about the rest of the tools needed to compile open_agb_firm? Will they work on Windows?

Reply:
The point of the comment was actually to point out to use the devkitPro supplied msys2 instead of WSL.The Linux variant of 7-Zip is probably available in msys2. Didn't check.

Reply:
Slightly unrelated, but in my experience, Linux was always the preferred platform for devkitPro, with many recommending Windows users to install WSL2 in order to use it. The Linux devkitPro install guide even mentions a workaround for WSL1 (which is nowhere near as good as WSL2).

I think mentioning WSL2 for Windows 10 users isn't a bad thing since the compile guide and tools definitely seem to be intended for Linux platforms. And while there is a `p7zip` package for MSYS2, there's no `p7zip-full` package, and I think the `p7zip` package only contains `7zr`, so we would need to adjust the Makefile.

Reply:
Unless they recommend WSL(2) i would still not do so here in this repo. You have not seen to which length they go to force people to use their recommended environment/tools. If you use anything else they go full insult/harass mode should you ever require help.

Reply:
I see. Okay, I'll remove the recommendation. I still feel like there should be something that recommends a Linux environment though.

Reply:
Fixed.